### ISSUE TITLE
only use a subset of user-installed fonts to farble

### ISSUE THREAD

**Issue by arthuredelstein** [brave/brave-browser#34043]
We are going to constrain font farbling to only those that are present on the fingerprint.js script.

**Comment by stephendonner** [comment]
Hi @arthuredelstein !  Mind adding a testplan here, when you get a chance?  Thanks!  Adding `QA/Blocked` and `QA/Test-Plan-Required` just until we've got that, and then we'll be fully unblocked 👍

**Comment by arthuredelstein** [comment]
**QA test plan:** Please confirm that this test still behaves as expected: https://dev-pages.brave.software/fingerprinting/fonts.html
Also it may be useful to repeat the test with a difference preferred language.

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```
Brave | 1.62.73 Chromium: 119.0.6045.163 (Official Build) nightly (x86_64)
-- | --
Revision | 522e9147d931744b1641084046c197caf7b341f0
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Steps:
1. installed `1.62.73`
2. launched Brave
3. loaded `https://dev-pages.brave.software/fingerprinting/fonts.html`
4. followed the instructions from the above page
5.

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using
```
Brave | 1.62.100 Chromium: 120.0.6099.71 (Official Build) beta (64-bit)
-- | --
Revision | ac084eaa0f81aa9ef631c2ca81a02469a4ee1a4a
OS | Windows 11 Version 22H2 (Build 22621.2715)
```


### Steps:
1. installed `1.62.100`
2. launched Brave
3. loaded `https://dev-pages.brave.software/fingerprinting/fonts.html`
4. followed the instructions from the above page (installed `fredoka one` via github; the font is no longer available at the link provided)
5. clicked `Start tests`
6. confirmed the `Control` and `Test` widths of the sample string `mmmmmmmmmmlli` are the same
7. clicked on the `Other site` link at `https://dev-pages.brave.software/fingerprinting/fonts.html`

### Confirmed both sets of test pages pass

example | example | example
------ | ------ | -------
<img width="582" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/ad5c8dc4-bc1b-49bd-95aa-8b412516d866">|<img width="958" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/cdfc8faf-cfe5-4f1e-93ec-dda864b098f0">|<img width="959" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/1d19a091-9b37-4112-bdc9-ac50f3da1d2f">

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 14 AP11.231020.013.A1 release-keys
Brave build: 1.62.130
Chromium: 120.0.6099.199 (Official Build) beta (64-bit) 
```
Filed follow-up issue #35188
---------------------------

**STEPS:**
   1. Use `Samsung` device
   2. Launch `Chrome` > Go to `https://arthuredelstein.net/demos/font.html`
   3. Confirm that `Attempting to load Samsung Color Emoji` uses Samsung emoji design
   4. Launch `Brave` > Go to `https://arthuredelstein.net/demos/font.html`
   5. Disable `Brave Shields` > Confirm that `Attempting to load Samsung Color Emoji` uses Samsung emoji design
   6. Now enable `Brave shields` > Refresh the page > Observe, that Brave still uses `Samsung` emoji design instead of `Google Noto Emoji` design

**ACTUAL RESULTS:**
- Observed that font restrictions are not getting applied on Android devices.

---------------------------

|Chrome|Brave. Shields OFF|Brave. Shields ON|
|:-:|:-:|:-:|
| ![1](https://github.com/brave/brave-browser/assets/128532965/bc5ac0e0-ec8d-4bb7-9874-0f618a8dfbd9) | ![2](https://github.com/brave/brave-browser/assets/128532965/7189807b-b2c2-4113-9976-debedf371443) | ![3](https://github.com/brave/brave-browser/assets/128532965/d5015ad7-54c8-438e-9c3d-9c7650ee08a4) |
